cypress/e2e/chapters/[chapterId].cy.ts

```diff
@@ -8,39 +7,24 @@ describe('chapter page', () => {
     cy.task('seedDb');
   });
 
+  //ToDo: check for if UI components are functioning as intended
```

Comment:
That's what tests were doing before. Why remove them, instead of updating?

Reply:
This UI is for MVP, we can have them back but I am going to style them in post MVP, and create switch as described in the subscribe issue in the profile, and I want the file to test both layer.

Reply:
Could you elaborate a bit? This sounds like tests were removed, because in future style will be changed

Reply:
I will update the comment with more details 👍

Reply:
I see the changed comment, but that doesn't explain it for me. Most of the UI, if not all, can be considered as not final (or maybe it is?), that doesn't mean all tests should be deleted.

The new tests are checking only if request to server doesn't result in error. The lack of error is not very strong proof that functionality is working. Another thing is by using only direct server request tests don't check the user side of things, user don't communicate with server directly.

Reply:
Sorry for the lack or reply, didn't notice the comment.

I don't remember specific reason of why I didn't implement cypress tests, and not update them. But I can tell you how I currently feel about adding test to these component.

I don't think there is need to cypress test the component, because the time spend going around the test and making the test work isn't worth it. 

We are using variables like `chapterUser?.subscribed`, which indicates if the user is joined or not to. 

So unless someone changes the variables' name, create pull request that passes enlist and build errors, pr is merged it, create conflict with someone else, then the 2nd person fix the conflict but doesn't account for changed variables, and passes enlist and build errors, then the broken pr is merged with flawed logic, and cause the button to not work.

Therefor, the variables like `chapterUser?.subscribed` should be acting as our static tests.

The users should be using join and leave chapter regularly, if this doesn't work we will know quickly about it.

Server issues are harder to fix, than client side, if this breaks in the client, I can't think of a scenario in which we can't revert the breaking change, fix it and implements new change in small period of time. (this may be me lacking experience in these parts)

---


Saying all this, although I find test around specific sequence or UI function annoying, I see it's usage. I am totally okay with update them and added them back if deemed needed

Reply:
nvm that oliver implement a test 🤣